cpp/oneapi/dal/backend/primitives/common.hpp

```diff
@@ -17,3 +17,13 @@
 #pragma once
 
 #include "oneapi/dal/backend/common.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+/// Specifies whether an input matrix should be transposed
+enum class transpose {
+    nontrans, /* do not transpose the matrix */
+    trans /* transpose the matrix */
+};
```

Comment:
1) Why it is specifically `enum class`?
2) Please specify underlying type: 
```suggestion
enum transpose : std::int32_t {
    nontrans, /* do not transpose the matrix */
    trans /* transpose the matrix */
};
```